outlay/app/src/test/java/app/outlay/example/LoginViewPresenterTest.java

```diff
@@ -0,0 +1,147 @@
+
+    @Captor
+    private ArgumentCaptor<DefaultSubscriber> SubscriberCaptor;
+
+    @Before
+    public void init(){
+        loginViewPresenter = new LoginViewPresenter(userSignInUseCaseMock, userSignUpUseCaseMock, resetPasswordUseCaseMock, linkAccountUseCaseMock);
+        loginViewPresenter.attachView(loginViewMock);
+
+        user.setEmail(userEmail);
+        user.setId("1");
+    }
+
+    @Test
+    public void public_signIn(){
+        loginViewPresenter.signIn(userEmail, userPassword);
+
+        //private signIn indirect test
+        verify(loginViewMock).setProgress(true);
+
+        verify(userSignInUseCaseMock).execute(any(),UserSubscriberCaptor.capture());
+        UserSubscriberCaptor.getValue().onNext(user);
+
+        //private onAuthSuccess indirect test
+        verify(loginViewMock).onSuccess(eq(user));
+
+        UserSubscriberCaptor.getValue().onError(e);
+        verify(loginViewMock).setProgress(false);
+        verify(loginViewMock).error(eq(e));
+    }
+
+    @Test
+    public void signInGuest_Test(){
+        loginViewPresenter.signInGuest();
+
+        //private signIn indirect test
+        verify(loginViewMock).setProgress(true);
+        verify(userSignInUseCaseMock).execute(eq(Credentials.GUEST),any());
+    }
+
+    @Test
+    public void signUp_Test(){
+        loginViewPresenter.signUp(userEmail,userPassword);
+
+        verify(loginViewMock).setProgress(true);
+
+        verify(userSignUpUseCaseMock).execute(any(),UserSubscriberCaptor.capture());
+        UserSubscriberCaptor.getValue().onNext(user);
+
+        //private onAuthSuccess indirect test
+        verify(loginViewMock).onSuccess(eq(user));
+
+        UserSubscriberCaptor.getValue().onError(e);
+        verify(loginViewMock).setProgress(false);
+        verify(loginViewMock).error(eq(e));
+    }
+
+    @Test
+    public void linkAccount_UsesLinkAccountUseCase(){
+        loginViewPresenter.linkAccount(userEmail,userPassword);
+
+        verify(linkAccountUseCaseMock).execute(any(),any());
+    }
+
+    @Test
+    public void linkAccount_DefaultSubscribe_OnNext(){
+        loginViewPresenter.linkAccount(userEmail,userPassword);
+
+        verify(linkAccountUseCaseMock).execute(any(), UserSubscriberCaptor.capture());
+        UserSubscriberCaptor.getValue().onNext(user);
+        verify(loginViewMock).onSuccess(eq(user));
+    }
+
+    @Test
+    public void resetPassword_UsesResetPasswordUseCase(){
+        loginViewPresenter.resetPassword(userEmail);
+
+        verify(resetPasswordUseCaseMock).execute(any(),any());
+    }
+
+    @Test
+    public void linkAccount_DefaultSubscribe_OnCompleted(){
+        loginViewPresenter.resetPassword(userEmail);
+
+        verify(resetPasswordUseCaseMock).execute(any(),SubscriberCaptor.capture());
+        SubscriberCaptor.getValue().onCompleted();
+        verify(loginViewMock).info(any());
+    }
+    @Test
+    public void linkAccount_DefaultSubscribe_onError(){
```

Comment:
Name has a copy-paste error: this is actually a test of the `resetPassword` method, and not `linkAccount`.